tests/plugin_filter/util.py

```diff
@@ -0,0 +1,218 @@
+        plugin.tags = tags
+        plugin.description = description
+        tmp_db.session.commit()
+        plugin_id = (
+            tmp_db.session.query(RAMP.id)
+            .filter(RAMP.name == name, RAMP.version == version)
+            .first()[0]
+        )
+        return plugin_id
+    plugin = RAMP(
+        name=name,
+        description=description,
+        version=version,
+        tags=tags,
+    )
+    plugin.plugin_id = f"{name}@{version}"
+    tmp_db.session.add(plugin)
+    tmp_db.session.commit()
+    plugin_id = (
+        tmp_db.session.query(RAMP.id)
+        .filter(RAMP.name == name, RAMP.version == version)
+        .first()[0]
+    )
+    return plugin_id
+
+
+def create_plugins(
+    tmp_db, tags: list[PluginTag], versions: list[str] = ["1.0.0", "1.1.0", "2.0.0"], n=3
+) -> None:
+    for i in range(n):
+        create_or_update_plugin(
+            tmp_db,
+            name=f"p{i}",
+            description=f"d{i}",
+            version=f"{random.choice(versions)}",
+            tags=[]
+            if not tags
+            else random.sample(tags, random.randint(1, len(tags) - 1)),
+        )
+
+
+def create_template_tab(
+    tmp_db,
+    client,
+    template_id: str = "1",
+    tab_name: str = "test_tab",
+    tab_description: str = "test tab",
+    tab_location: str = "test_location",
+    sort_key: int = 0,
+    filter_dict: dict | None = None,
+) -> int:
+    if not filter_dict:
+        filter_dict = {}
+    filter_string = json.dumps(filter_dict)
+    response = client.post(
+        f"/api/templates/{template_id}/tabs/",
+        json={
+            "name": tab_name,
+            "description": tab_description,
+            "location": tab_location,
+            "sortKey": sort_key,
+            "filterString": filter_string,
+        },
+    )
+    assert response.status_code == 200
+    template_tab_id = (
+        tmp_db.session.query(TemplateTab.id)
+        .filter(
+            TemplateTab.name == tab_name,
+            TemplateTab.description == tab_description,
+            TemplateTab.location == tab_location,
+            TemplateTab.sort_key == sort_key,
+            TemplateTab.filter_string == filter_string,
+        )
+        .first()[0]
+    )
+    apply_filter_for_tab.apply(args=(template_tab_id,))
+    return template_tab_id
+
+
+def create_tags(tmp_db, tags: list[str] | None = None) -> list[PluginTag]:
+    if not tags:
+        plugin_tags = [PluginTag(f"t{tag}") for tag in range(5)]
+        tags = [tag.tag for tag in plugin_tags]
+    else:
+        plugin_tags = [PluginTag(tag) for tag in tags]
+    for tag in plugin_tags:
+        tag_exists = tmp_db.session.query(
+            exists().where(PluginTag.tag == tag.tag)
+        ).scalar()
+        if not tag_exists:
+            tmp_db.session.add(tag)
+    tmp_db.session.commit()
+    return tmp_db.session.query(PluginTag).filter(PluginTag.tag.in_(tags)).all()
+
+
+def compare_plugins(plugin: RAMP, other: RAMP):
```

Comment:
For most tests it should suffice to just check that the database id (`plugin.id`) is the same.